x/spec/README.md

```diff
@@ -0,0 +1,281 @@
+
+This document is focuses on the spec' technical aspects. For more information on chain support in lava see pairing README.
+
+## Contents
+* [Concepts](#concepts)
+  * [Spec](#spec)
+  * [ApiCollections](#apicollections)
+  * [CollectionData](#collectiondata)
+  * [ParseDirective](#parsedirective)
+  * [Verification](#verification)
+  * [Import](#import)
+* [Parameters](#parameters)
+* [Queries](#queries)
+* [Transactions](#transactions)
+* [Proposals](#proposals)
+
+## Concepts
+
+### Spec
+
+A chain spec consists of general properties of the chain and a list of interface it supports. To make specs simple to create and maintain specs can import api's from another, for example, X testnet spec can Import from X mainnet spec and thus not need to redifine all of the interfaces.
+
+```
+type Spec struct {
+	Index                         string                // chain index (index key for the chain)                             
+	Name                          string                // description string of the spec
+	Enabled                       bool                  // spec enabled/disable (base specs are usually disabled, like cosmos base specs)
+	AverageBlockTime              int64                 // average block time of the chain in msec
+	MinStakeProvider              Coin                  // min stake for a provider to be active in the chain
+	ProvidersTypes                Spec_ProvidersTypes   // determines if the spec is for lava or others 
+	Imports                       []string              // list of chains to import ApiCollections from
+	ApiCollections                []*ApiCollection      // list of ApiCollections that defines all the interfaces and api's of the chain
+	Contributor                   []string              // list of contributers (public lava address {lava@...})
+	ContributorPercentage         *Dec                  // the percentage of coins the contributers will get from each reward a provider get
+	Shares                        uint64                // factor for bonus rewards at the end of the month (see rewards module)
+    AllowedBlockLagForQosSync     int64                 // defines the accepted blocks a provider can be behind the chain without QOS degradation
+	BlockLastUpdated              uint64                // the last block this spec was updated on chain
+    ReliabilityThreshold          uint32                // this determines the amount of data reliability for the chain
+	DataReliabilityEnabled        bool                  // enables/disables data reliability for the chain
+	BlockDistanceForFinalizedData uint32                              
+	BlocksInFinalizationProof     uint32                              
+}
+```
+Note, the `Coin` type is from Cosmos-SDK (`cosmos.base.v1beta1.Coin`).
+Note, the `Dec` type is from Cosmos-SDK math LegacyDec.
```

Comment:
put extra line between the "Note" sentences to make it appear as a new line. Also, in the second "Note" I would replace it with: "Also" or "Also note"